### PR TITLE
Mttl 886 raise deleted event

### PR DESCRIPTION
**Pull request by evak6979**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-886

### *What is the problem we're trying to solve*

Raise an event when a contact is deleted.

### CODE REVIEW

**Review by humulla — APPROVED** [review]

File: ContactDetailsApi/V1/Controllers/ContactDetailsController.cs [hunk 1]
```diff
@@ -69,8 +70,9 @@ public async Task<IActionResult> GetContactDetailsByTargetId([FromQuery] Contact
         [LogCall(LogLevel.Information)]
         public async Task<IActionResult> DeleteContactDetailsById([FromQuery] DeleteContactQueryParameter queryParam)
         {
+            var token = _tokenFactory.Create(_httpContextWrapper.GetContextRequestHeaders(HttpContext));
 
-            var contact = await _deleteContactDetailsByTargetIdUseCase.Execute(queryParam).ConfigureAwait(false);
+            var contact = await _deleteContactDetailsByTargetIdUseCase.Execute(queryParam, token, ContactDetailsConstants.DELETED).ConfigureAwait(false);
```
Comment (mikemoody-amido) [inline ContactDetailsApi/V1/Controllers/ContactDetailsController.cs:75]: Not sure why you pass the event type constant from here. Surely you can not put it here and just specify in the use case (as each use case is specific to the event type)?

File: ContactDetailsApi/V1/UseCase/CreateContactUseCase.cs [hunk 2]
```diff
@@ -22,22 +24,17 @@ public CreateContactUseCase(IContactDetailsGateway gateway, ISnsGateway snsGatew
             _snsFactory = snsFactory;
         }
 
-        public async Task<ContactDetailsResponseObject> ExecuteAsync(ContactDetailsRequestObject contactRequest, Token token)
+        public async Task<ContactDetailsResponseObject> ExecuteAsync(ContactDetailsRequestObject contactRequest,
+            Token token, string eventType)
```
Comment (mikemoody-amido) [inline ContactDetailsApi/V1/UseCase/CreateContactUseCase.cs:28]: Why pass the event type in? This use case is specific already...

File: ContactDetailsApi/V1/UseCase/DeleteContactDetailsByTargetIdUseCase.cs [hunk 3]
```diff
@@ -1,26 +1,50 @@
...
+        public DeleteContactDetailsByTargetIdUseCase(IContactDetailsGateway gateway, ISnsFactory factory, ISnsGateway snsGateway)
         {
             _gateway = gateway;
+            _snsFactory = factory;
+            _snsGateway = snsGateway;
         }
 
         [LogCall]
-        public async Task<ContactDetailsResponseObject> Execute(DeleteContactQueryParameter query)
+        public async Task<ContactDetailsResponseObject> Execute(DeleteContactQueryParameter query, Token token, string eventType)
```
Comment (mikemoody-amido) [inline ContactDetailsApi/V1/UseCase/DeleteContactDetailsByTargetIdUseCase.cs:29]: Why pass the event type in? This use case is specific already...

File: ContactDetailsApi/V1/Controllers/ContactDetailsController.cs (hunk 1 above)
Comment (evak6979) [reply ContactDetailsApi/V1/Controllers/ContactDetailsController.cs:75]: Good point, I can revert that.

File: ContactDetailsApi/V1/UseCase/CreateContactUseCase.cs (hunk 2 above)
Comment (evak6979) [reply ContactDetailsApi/V1/UseCase/CreateContactUseCase.cs]: Because that's the way I like it ! (But even so, fixed it :P )

**Review by mikemoody-amido — APPROVED** [review]

**Pull request merged**